client/components/editor.vue

```diff
@@ -597,43 +609,93 @@
             }
             // update collection to reflect new segments of the selection with previous highlight marks
             newChunks = updateChunks(newChunks, chunkToSplit, {from: start, to: end});
-            const currentKeys = marks[0].attrs["highlight-keys"];
-            removeHighlight(schema, this.$editorView.state, start, end, this.$editorView.dispatch);
-            addHighlight(`${currentKeys},${key}`, schema, this.$editorView.state, start, end, this.$editorView.dispatch);
+            const currentKeys = marks[0].attrs['highlight-keys'];
+            removeHighlight(schema, tr, doc, start, end, this.$editorView.dispatch);
+            let keys = `${this.commentHighlightKey},${currentKeys}`;
+            // If the new highlight contains the initial part of another highlight.
+            if (selection.from < start) {
+              keys = `${currentKeys},${this.commentHighlightKey}`;
```

Comment:
Why the order of these highlight keys matter?

Reply:
When the cursor is in a particular position where there are multiple highlight marks, we need a way to identify the comment on the sidebar that should be focused (like Google Docs does). So, thats why I added this condition on highlight creation. The order of the keys is important because the first key in the list corresponds to the highlight key of the comment that should be focused.